presto-main/src/main/java/com/facebook/presto/sql/planner/sanity/NoExpressionLeftCheck.java

```diff
@@ -0,0 +1,151 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.sql.planner.sanity;
+
+import com.facebook.presto.Session;
+import com.facebook.presto.execution.warnings.WarningCollector;
+import com.facebook.presto.metadata.Metadata;
+import com.facebook.presto.spi.plan.FilterNode;
+import com.facebook.presto.spi.plan.PlanNode;
+import com.facebook.presto.spi.plan.ValuesNode;
+import com.facebook.presto.spi.relation.RowExpression;
+import com.facebook.presto.sql.parser.SqlParser;
+import com.facebook.presto.sql.planner.SimplePlanVisitor;
+import com.facebook.presto.sql.planner.TypeProvider;
+import com.facebook.presto.sql.planner.plan.AggregationNode;
+import com.facebook.presto.sql.planner.plan.JoinNode;
+import com.facebook.presto.sql.planner.plan.ProjectNode;
+import com.facebook.presto.sql.planner.plan.SpatialJoinNode;
+import com.facebook.presto.sql.planner.plan.WindowNode;
+
+import java.util.List;
+
+import static com.facebook.presto.sql.relational.OriginalExpressionUtils.isExpression;
+
+/**
+ * Ensures that FilterNode and ValuesNode have no OriginalExpression left
+ */
+public class NoExpressionLeftCheck
+        implements PlanSanityChecker.Checker
+{
+    static final String ERROR_MESSAGE = "OriginalExpression still exists in logical plan";
+
+    @Override
+    public void validate(PlanNode plan, Session session, Metadata metadata, SqlParser sqlParser, TypeProvider types, WarningCollector warningCollector)
+    {
+        validate(plan);
+    }
+    public static void validate(PlanNode plan)
```

Comment:
`@VisibleForTest`

Reply:
Is this function really necessary? Can tests just use the proper API?

Reply:
> Is this function really necessary? Can tests just use the proper API?

Yes, this problem will be fixed